microcosm_postgres/diff.py

```diff
@@ -0,0 +1,39 @@
+"""
+Compute ORM differences.
+
+"""
+from itertools import chain
+
+from sqlalchemy.orm import class_mapper, ColumnProperty
+
+
+class Version(dict):
+    """
+    Capture of object state a specific time.
+
+    """
+    def __init__(self, instance):
+        super(Version, self).__init__({
+            prop.key: getattr(instance, prop.key)
+            for prop in class_mapper(instance.__class__).iterate_properties
+            if isinstance(prop, ColumnProperty)
+        })
+
+    def difference(self, other):
+        return Delta(self, other)
+
+    def __sub__(self, other):
+        return Delta(self, other)
+
+
+class Delta(dict):
+    """
+    Difference beween two object versions.
+
+    """
+    def __init__(self, left, right):
+        super(Delta, self).__init__({
+            key: right.get(key)
```

Comment:
I think for the state transition case we care specifically about the (from, to) tuple (e.g (old, new) values) so would be valuable there if this ends up powering some of that